src/rpc/blockchain.cpp

```diff
@@ -1289,7 +1291,8 @@ RPCHelpMan getblockchaininfo()
     obj.pushKV("pruned", chainman.m_blockman.IsPruneMode());
     if (chainman.m_blockman.IsPruneMode()) {
         bool has_tip_data = tip.nStatus & BLOCK_HAVE_DATA;
```

Comment:
done, I liked that `BLOCK_HAVE_DATA | BLOCK_HAVE_UNDO` is more expressive because we don't use `BLOCK_HAVE_MASK` much I feel like people would need to look it up more, but I have just added a comment instead now.